### PR TITLE
[FrameworkBundle] simplify dep declaration

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

I'm not sure this will work, travis will tell.

### CODE REVIEW

**Comment by stof** [comment]
this is indeed equivalent

**Comment by nicolas-grekas** [comment]
I wondered if mentioning a non existing version (2.3.26/2.6.5) would make composer fail.
Look like not, good to merge then.

File: src/Symfony/Bundle/FrameworkBundle/composer.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
         "symfony/http-foundation": "~2.4.9|~2.5,>=2.5.4",
         "symfony/http-kernel": "~2.6",
         "symfony/filesystem": "~2.3",
-        "symfony/routing": "~2.3.25,>2.3.25|~2.6,>2.6.4",
+        "symfony/routing": "~2.3.26|~2.6.5|~2.7",
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]: Don't we have other similar examples in other components which could be simplified as well?

Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]: I did not find any other

Comment (jeremy-derusse) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]: Why do you convert the part `~2.6,>2.6.4` into `~2.6.5|~2.7` but you don't convert `symfony/http-foundation` into `~2.4.9|~2.5.4|~2.6`? or `symfony/dependency-injection` into `~2.6.2|~2.7` ?

Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]: I do use the same style as the one for symfony/http-foundation now

**Comment by nicolas-grekas** [comment]
All the ~2.X,>=2.X.Y will be replaceable by ^2.X.Y once January’s version of composer becomes mainstream and the caret constraints are usable.

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**